backend-rust/migrations/0001_initialize.up.sql

```diff
@@ -247,6 +247,63 @@ CREATE TABLE bakers(
         BIGINT
 );
 
+-- Every WASM module on chain.
+CREATE TABLE smart_contract_modules(
+    index
+        BIGINT
+        PRIMARY KEY,
+    -- Module reference of the wasm module.
+    module_reference
+        CHAR(64)
+        UNIQUE
+        NOT NULL,
+    -- The absolute block height when the module was deployed.
+    deployment_block_height
+        BIGINT
+        NOT NULL,
+    -- Transaction index in the block deploying the module.
+    deployment_transaction_index
+        BIGINT
+        NOT NULL,
+    -- Embedded schema in the wasm module if present.
+    schema TEXT
+);
+
+-- Every contract instance on chain.
+CREATE TABLE contracts(
+    -- Index of the contract.
+    index
+        BIGINT
+        NOT NULL,
+    -- Sub index of the contract.
+    sub_index
+        BIGINT
+        NOT NULL,
+    -- Note: It might be better to use `module_reference_index` which would save storage space but would require more work in inserting/querying by the indexer.
+    -- Module reference of the wasm module.
+    module_reference
+        CHAR(64)
+        NOT NULL,
+    -- The contract name.
+    name
+        TEXT
+        NOT NULL,
+    -- The total balance of the contract in micro CCD.
+    amount
+        BIGINT
+        NOT NULL,
+    -- The absolute block height when the module was initialized.
+    init_block_height
+        BIGINT
+        NOT NULL,
+    -- Transaction index in the block initializing the contract.
+    init_transaction_index
+        BIGINT
+        NOT NULL,
+
+    -- Make the contract index and subindex the primary key.
```

Comment:
don't see a reason why that needs to be removed.

Reply:
The comment does not say anything that is not already said by the code. If you wanted to add a comment here, it would make more sense to add _why_ the index and subindex are the primary key, for example.

Reply:
It does :). Since `index`/`subindex` can refer to several different things, the comments makes it clear that we talk about `contract indexes here`.

Reply:
I don't think it does. It is clear that these are contract indices since this is the `contracts` table. `index` and `subindex` could not refer to anything else than the columns of this table (you cannot refer to anything but columns of the table you're defining, anything else is out of scope).

Anyways, it is not important, but do remember to write meaningful comments that add on top of the code, not just repeating the code :)